src/liballoc_jemalloc/build.rs

```diff
@@ -111,7 +111,7 @@ fn main() {
     println!("cargo:rustc-link-search=native={}/lib", build_dir.display());
     if target.contains("android") {
         println!("cargo:rustc-link-lib=gcc");
-    } else if !target.contains("windows") {
+    } else if !target.contains("windows") && !target.contains("musl") {
```

Comment:
Updated logic to match non-rustbuild builds. See https://github.com/rust-lang/rust/blob/c116ae35cf49b55bd8d82e31f1ba030cf7e63867/src/liballoc_jemalloc/lib.rs#L37-L40